mycroft/util/parse.py

```diff
@@ -164,6 +189,623 @@ def extractnumber_en(text):
+                        remainder = "am"
+                        used = 2
+                    elif wordNext == "this" and wordNextNext == "afternoon":
+                        remainder = "pm"
+                        used = 2
+                    elif wordNext == "this" and wordNextNext == "evening":
+                        remainder = "pm"
+                        used = 2
+                    elif wordNext == "at" and wordNextNext == "night":
+                        if strHH > 5:
+                            remainder = "pm"
+                        else:
+                            remainder = "am"
+                        used += 2
+                    else:
+                        if timeQualifier != "":
+                            military = True
+                            if strHH <= 12 and \
+                                    (timeQualifier == "evening" or
+                                        timeQualifier == "afternoon"):
+                                strHH += 12
+            else:
+                # try to parse # s without colons
+                # 5 hours, 10 minutes etc.
+                length = len(word)
+                strNum = ""
+                remainder = ""
+                for i in range(length):
+                    if word[i].isdigit():
+                        strNum += word[i]
+                    else:
+                        remainder += word[i]
+
+                if remainder == "":
+                    remainder = wordNext.replace(".", "").lstrip().rstrip()
+                if remainder == "pm" or wordNext == "pm" or \
+                        remainder == "p.m." or wordNext == "p.m.":
+                    strHH = strNum
+                    remainder = "pm"
+                    used = 1
+                elif remainder == "am" or wordNext == "am" or \
+                        remainder == "a.m." or wordNext == "a.m.":
+                    strHH = strNum
+                    remainder = "am"
+                    used = 1
+                else:
+                    if wordNext == "pm" or wordNext == "p.m.":
+                        strHH = strNum
+                        reaminder = "pm"
+                        used = 1
+                    elif wordNext == "am" or wordNext == "a.m.":
+                        strHH = strNum
+                        remainder = "am"
+                        used = 1
+                    elif int(word) > 100 and \
+                            (wordPrev == "o" or wordPrev == "oh"):
+                        # 0800 hours (pronounced oh-eight-hundred)
+                        strHH = int(word)/100
+                        strMM = int(word) - strHH*100
+                        military = True
+                        if wordNext == "hours":
+                            used += 1
+                    elif wordNext == "hours" and word[0] != '0' and \
+                            (int(word) < 0100 and int(word) > 2400):
```

Comment:
I actually did not know it would interpret that as octal. I just figured it would be more clear that this is intended to parse military time. Will fix.